Networking/Networking/Model/Product/ProductCategory.swift

```diff
@@ -5,29 +5,41 @@ import Foundation
 ///
 public struct ProductCategory: Decodable {
     public let categoryID: Int64
+    public let siteID: Int64
+    public let parentID: Int64
     public let name: String
     public let slug: String
 
     /// ProductCategory initializer.
     ///
     public init(categoryID: Int64,
+                siteID: Int64,
+                parentID: Int64,
                 name: String,
                 slug: String) {
         self.categoryID = categoryID
+        self.siteID = siteID
+        self.parentID = parentID
         self.name = name
         self.slug = slug
     }
 
     /// Public initializer for ProductCategory.
     ///
     public init(from decoder: Decoder) throws {
+        guard let siteID = decoder.userInfo[.siteID] as? Int64 else {
+            throw ProductCategoryDecodingError.missingSiteID
+        }
+
         let container = try decoder.container(keyedBy: CodingKeys.self)
 
         let categoryID = try container.decode(Int64.self, forKey: .categoryID)
+        // Some product endpoints don't include the parent category ID
```

Comment:
Oh yeah, good point!
I've added some code to prevent it and a test for it! https://github.com/woocommerce/woocommerce-ios/pull/2079/commits/6edc40065647a50b02dcc12e11c31f07d538ea93

Reply:
Great! Thanks for adding the test too. That's awesome!